client/src/pages/Dashboard/DashboardPage.tsx

```diff
@@ -1,11 +1,394 @@
-import { Typography } from "antd";
-import React from "react";
+/* eslint-disable no-plusplus */
+import {
+  CalendarOutlined,
+  CarryOutOutlined,
+  CheckOutlined,
+  CloseOutlined,
+  EditOutlined,
+} from "@ant-design/icons";
+import {
+  Col,
+  DatePicker,
+  Divider,
+  Form,
+  Input,
+  Modal,
+  Popover,
+  Row,
+  Timeline,
+  Typography,
+} from "antd";
+import moment from "moment";
+import React, { useState } from "react";
+
+import DashboardCard from "../../components/DashboardCard/DashboardCard";
+import { syncs as timelineSyncs } from "../Families/FamiliesPages";
+import { UpcomingSyncType } from "../types";
 
 const { Title } = Typography;
 
-const DashboardPage = (): JSX.Element => (
-  <>
-    <Title>Dashboard!</Title>
-  </>
-);
+enum EditModalEnum {
+  SUMMARY = "Summary",
+  RESCHEDULE = "Reschedule",
+}
+
+type DashboardPageState = {
+  syncs: UpcomingSyncType[];
+  isModalVisible: boolean;
+  editingSync: UpcomingSyncType;
+  editingMode: EditModalEnum;
+  editingInfo: {
+    title: string;
+    details: string;
+    date: Date;
+  };
+};
+
+let id = 0;
+const today = new Date(2021, 9, 15); // month is 0-index, equivalent to "2021-10-15"
+const initialSyncs: UpcomingSyncType[] = [
+  {
+    id: id++,
+    name: "Peter",
+    family: "Twelves",
+    lastSynced: new Date(2021, 8, 1),
+    upcomingSync: new Date(2021, 9, 1),
+  },
+  {
+    id: id++,
+    name: "Andrew",
+    family: "Twelves",
+    lastSynced: new Date(2021, 8, 1),
+    upcomingSync: new Date(2021, 9, 4),
+  },
+  {
+    id: id++,
+    name: "James",
+    family: "Twelves",
+    lastSynced: new Date(2021, 8, 1),
+    upcomingSync: new Date(2021, 9, 5),
+  },
+  {
+    id: id++,
+    name: "John",
+    family: "Twelves",
+    lastSynced: new Date(2021, 8, 1),
+    upcomingSync: new Date(2021, 9, 8),
+  },
+  {
+    id: id++,
+    name: "Bartholomew",
+    family: "Twelves",
+    lastSynced: new Date(2021, 8, 1),
+    upcomingSync: new Date(2021, 9, 10),
+  },
+  {
+    id: id++,
+    name: "Matthew",
+    family: "Twelves",
+    lastSynced: new Date(2021, 8, 1),
+    upcomingSync: new Date(2021, 9, 16),
+  },
+  {
+    id: id++,
+    name: "Thomas",
+    family: "Twelves",
+    lastSynced: new Date(2021, 8, 1),
+    upcomingSync: new Date(2021, 9, 18),
+  },
+  {
+    id: id++,
+    name: "Simon",
+    family: "Twelves",
+    lastSynced: new Date(2021, 8, 1),
+    upcomingSync: new Date(2021, 9, 20),
+  },
+  {
+    id: id++,
+    name: "Judas",
+    family: "Twelves",
+    lastSynced: new Date(2021, 8, 1),
+    upcomingSync: new Date(2021, 9, 20),
+  },
+  {
+    id: id++,
+    name: "Philip",
+    family: "Twelves",
+    lastSynced: new Date(2021, 8, 1),
+    upcomingSync: new Date(2021, 9, 17),
+  },
+];
```

Comment:
Feel like this should rather live in the test.ts or a seperate fixtures.ts (https://stackoverflow.com/questions/12071344/what-are-fixtures-in-programming)